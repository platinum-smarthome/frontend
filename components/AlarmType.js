import React, { Component } from 'react';
import { View, Image, Text, TouchableOpacity, StyleSheet } from 'react-native'
import DoorStatus from './status/DoorStatus'

class AlarmType extends Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.status ? this.props.press : this.props.lock}>
        <View style={styles.alarmTypes}>
          <View>
            <Image style={styles.alarmImg} source={ this.props.status ? require('../components/assets/puerta.png') : require('../components/assets/access.png') }/>
          </View>
          <DoorStatus text={ this.props.text } status={ this.props.status }/>
          <TouchableOpacity style={styles.arrowPos}>
            <Image style={{ height: 30, width: 30}} source={ this.props.status ? require('../components/assets/greyarrow.png') : require('../components/assets/desbloqueado.png')} />
          </TouchableOpacity>
        </View>
      </TouchableOpacity>
    )
  }
}


const styles = StyleSheet.create({
  alarmTypes: {
    flexDirection: 'row',
    borderBottomWidth: 1,
    borderBottomColor: '#d3d3d3',
  },
  alarmImg: {
    width: 64,
    height: 64,
    marginHorizontal: 15,
    marginVertical: 6
  },
  textContent: {
    paddingVertical: 10
  },
  alarmText: {
    fontSize: 24,
    fontWeight: '300',
    color: '#333333'
  },
  arrowPos: {
    position: 'absolute',
    right: 0,
    paddingHorizontal: 25,
    paddingVertical: 20
  },
  locked: {
    fontSize: 16,
    fontWeight: '300',
    color: '#006ac6'
  },
  unlocked: {
    fontSize: 16,
    fontWeight: '300',
    color: 'red'
  }
})

export default AlarmType