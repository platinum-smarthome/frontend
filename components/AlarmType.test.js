import React, { Component } from 'react'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import AlarmType from './AlarmType'

import { Text, View, Image, TouchableOpacity } from 'react-native'

enzyme.configure({ adapter: new Adapter()})

let wrapper

beforeEach(() => {
  wrapper = shallow(<AlarmType status={true}/>)
})

describe('<AlarmType/>', () => {
  it('should render <AlarmType/>', () => {
    expect(wrapper).toBeDefined()
  })
})

describe('<AlarmType/> child is rendered', () => {
  it('should render <TouchableOpacity/>' , () => {
    expect(wrapper.find(<TouchableOpacity/>)).toBeDefined()
    expect(wrapper.find('TouchableOpacity').length).toBe(1)
  })
  it('Should render <View/>', () => {
    expect(wrapper.find(<View/>)).toBeDefined()
    expect(wrapper.find('View').length).toBe(3)  
  })
  it('Should render <Image/>', () => {
    expect(wrapper.find(<Image/>)).toBeDefined()
    expect(wrapper.find('Image').length).toBe(2)  
  })
})

describe('<Image/> renders the correct image', () => {
  it('should render the proper image if status is true', () => {
    let wraptrue = shallow(<AlarmType status={true}/>)
    expect(wraptrue.find(<Image source={require('../components/assets/puerta.png')}/>)).toBeDefined()
    expect(wraptrue.find(<Image source={require('../components/assets/greyarrow.png')}/>)).toBeDefined()
  })
  it('should render the proper image if status is true', () => {
    let wrapfalse = shallow(<AlarmType status={false}/>)
    expect(wrapfalse.find(<Image source={ require('../components/assets/access.png')} />)).toBeDefined()
    expect(wrapfalse.find(<Image source={ require('../components/assets/desbloqueado.png')} />)).toBeDefined()
  })
})

describe('snapshot testing', () => {
  it('snapshot', () => {
    const tree = renderer.create(<AlarmType/>)
    expect(tree).toMatchSnapshot()
  })
})