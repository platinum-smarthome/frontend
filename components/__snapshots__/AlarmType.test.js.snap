// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "borderBottomColor": "#d3d3d3",
        "borderBottomWidth": 1,
        "flexDirection": "row",
      }
    }
  >
    <View>
      <Image
        source={
          Object {
            "testUri": "../../../components/assets/access.png",
          }
        }
        style={
          Object {
            "height": 64,
            "marginHorizontal": 15,
            "marginVertical": 6,
            "width": 64,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "paddingVertical": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 24,
            "fontWeight": "300",
          }
        }
      >
         
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "red",
            "fontSize": 16,
            "fontWeight": "300",
          }
        }
      >
         Status: Unlocked 
      </Text>
    </View>
    <View
      style={
        Object {
          "paddingHorizontal": 25,
          "paddingVertical": 20,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../components/assets/desbloqueado.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</View>
`;
